### DISCUSSION TITLE
Security notice SN-2024-101 CVE-2024-6345 setuptools==v71.1.0

### DISCUSSION THREAD

**Discussion by ColruytGroupITMainframe** [ibm-z-oss-oda/python_ai_toolkit_zos#37] [General]
**Security notice SN-2024-101**
Python AI Toolkit Security updates
IBM Confidential
Last modified 2024-10-16
Summary
There is one vulnerability resolved in this update for an issue in a Toolkit package:
• CVE-2024-6345

**Vulnerability Details**
CVEID: CVE-2024-6345
Description: pypa/setuptools could allow a remote attacker to execute arbitrary code on the system, caused by an error in the package_index module. By persuading a victim to click a specially crafted URL, an attacker could exploit this vulnerability using its download functions to inject and execute arbitrary code on the system.
CVSS Base Score: 8.8
CVSS Vector: (CVSS:3.0/AV:N/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H)

**Affected Products and Versions**
Python AI Toolkit for IBM z/OS: 1.1.0

**Remediation/Fixes**
Product VRMF Remediation / Fix Location
Python AI Toolkit for IBM z/OS 1.1.0 The Python AI Toolkit for IBM z/OS is a library of Python functions that can be installed at runtime by a Python application. The fixes for the CVEs described in the Summary section provide the remediation for the full product. Users may install some or all the fixed packages described in this document as needed for their particular Python environment.
Member packages for the Python AI Toolkit for z/OS can be installed through the pip command from the pypi repository provided by IBM at (this is for use by pip only, it is not a browsable URL): https://downloads.pyaitoolkit.ibm.net/repository/python_ai_toolkit_zos/simple
You may install the following packages at the specified versions to obtain resolution for all the vulnerabilities (CVEs) listed above: **setuptools==v71.1.0**

The latest version available is in the toolkit is _setuptools-67.1.0-py3-none-any.whl_   is it possible to add a new version to handle the CVE report? 

With kind regards, 
Arthur Coucke 

Colruyt Group IT

**Comment by jbostian** [comment]
Hi Arthur,
     Unfortunately we can't comment on specific security issues here.  Security issues are handled through the IBM Z and LinuxONE Security Portal ([https://www.ibm.com/support/resourcelink/security-portal/resources](https://www.ibm.com/support/resourcelink/security-portal/resources)).  This interface will have information about specific CVE resolution as new versions of Toolkit packages are are released.

**Reply by ColruytGroupITMainframe** [reply]
@jbostian, I'm not discussing the CVE itself here, I'm asking when the setuptools in the Toolkit will be updated to v71.1.0.

**Comment by jbostian** [comment]
Our current policies for IBM Z don't allow us to comment on security issues, including expected timeframes for fixes.  We're only allowed to point to the security portal.

**Reply by ColruytGroupITMainframe** [reply]
I don't understand the problem. I will open another discussion and rephrase my question.

**Comment by ColruytGroupITMainframe** [comment]
New discussion will opened as this one seems to be confusing with the reference to the CVE
